src/promptflow/promptflow/batch/_csharp_executor_proxy.py

```diff
@@ -23,115 +28,52 @@ def create(
         *,
         connections: Optional[dict] = None,
         storage: Optional[AbstractRunStorage] = None,
-    ) -> "AbstractExecutorProxy":
+        **kwargs,
+    ) -> "CSharpExecutorProxy":
         """Create a new executor"""
-        working_dir = working_dir or flow_file.parent
-
-        from pythonnet import load
-
-        # pythonnet.load must be called before import clr
-        load("coreclr")
-        import clr
-
-        with _change_working_dir(working_dir):
-            sys.path.append(working_dir.as_posix())
-            clr.AddReference("Promptflow")
-            sys.path.pop()
-            from Promptflow.Executor import YamlExecutor
-
-            connection_provider_url = ""
-            executor = YamlExecutor(
-                flow_file.read_text(encoding="utf-8"), connection_provider_url, working_dir.as_posix()
-            )
-            return cls(executor)
+        port = cls.find_available_port()
+        log_path = kwargs.get("log_path", "")
+        command = [
+            "dotnet",
+            EXECUTOR_SERVICE_DLL,
+            "-p",
+            port,
+            "--yaml_path",
+            flow_file,
+            "--assembly_folder",
+            ".",
+            "--connection_provider_url",
```

Comment:
Where is connection_provider_url?

Reply:
For local we don't need to pass it to executor service. However, we have to pass an empty string to executor because it is a required param.

Reply:
why we don't need to pass it to executor service for local? csharp executor still depends on this to run its connection provider right?

Reply:
as offline synced, please add comment to explain that executor has done service resolution for now and remove this parameter after it's set to optional

Reply:
Added.